snmp/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,369 @@
+          This option works best when `oid_batch_size` is set to 128 or 256.
+          This is useful to improve performance when fetching metrics from large tables.
+          When OID caching is enabled, table row OIDs are reused. This means that the table can be fetched
+          using batched GET queries, as opposed to one GETNEXT query per row in the table.
+          The caching is disabled if the interval is set to `0` (default).
+          Caveat: If some tables rows are dynamic, new rows will only be fetched at every refresh interval.
+                  - This means new OIDs will only be picked up when OIDs are refreshed according to the refresh interval.
+                  - With auto discovery, if the same IP is assigned to a different device with different OIDs,
+                  the OIDs will be refreshed only according to the refresh interval.
+          Only available using python SNMP integration
+        
+        value:
+          type: integer
+          example: 3600
+        display_default: 0
+      - name: optimize_mib_memory_usage
+        description: |
+          **Experimental** - Setting this flag to true attempts to optimize memory usage related to
+          MIB information. This shares internal MIB data across instances.
+          Note: The absence of impact on MIB resolution for this strategy is not guaranteed yet.
+          Only available using python SNMP integration
+        value:
+          type: boolean
+          example: false
+          display_default: false
+      - name: mibs_folder
+        description: |
+          Specify an additional folder for your custom mib files (python format).
+          Only available using python SNMP integration
+        value:
+          type: string
+          example: /path/to/your/mibs/folder
+          display_default: 
+      - name: ignore_nonincreasing_oid 
+        description: |
+          Set to true to ignore non increasing oid.
+          See https://github.com/etingof/pysnmp/blob/master/docs/source/faq/oids-not-increasing.rst
+          Only available using python SNMP integration
+        value:
+          type: boolean
+          example: false
+          display_default: false
+      - template: init_config/default
+    - template: instances
+      options:
+      - name: ip_address
+        description: |
+          The IP address of the device to monitor.
+          One of ip_address and network_address must be provided.
+        value:
+          type: string
+          display_default:
+      - name: port
+        description: |
+          Default SNMP port.
+        value:
+          type: integer
+          example: 161
+          display_default: 161
+      - name: community_string
+        description: |
+          Only useful for SNMP v1 & v2.
+          Enclose the community string with single quote to avoid special character being interpreted.
+        value:
+          example: 'public'
+          type: string
+          display_default:
```

Comment:
```suggestion
          Enclose the community string with single quotes to avoid special characters being interpreted.
        value:
          default: public
          example: "'public'"
          type: string
```